### PR TITLE
[#152670228] Add go image

### PR DESCRIPTION
**Pull request by henrytk**
## What

This is a generic image for running Go programs. It contains `dep`
and `ginkgo`.

We initially intend to use this for the integration tests for the Elasticache broker (https://www.pivotaltracker.com/story/show/152670228)

## How to review

Check it is set up as per the README. Confirm the need to create such an image is justified.

## Who

Anyone but me or @camelpunch

### CODE REVIEW

File: go/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+FROM golang:1.9-alpine
+
+ADD https://github.com/golang/dep/releases/download/v0.3.2/dep-linux-amd64 /usr/local/bin/dep
+
+RUN chmod +x /usr/local/bin/dep
+
+RUN apk add --update git && rm -rf /var/cache/apk/*
+RUN go get github.com/onsi/ginkgo/ginkgo
```
[the remark this answers is not in the document]
Comment (henrytk) [reply go/Dockerfile]: We had a fist fight over it. @camelpunch agrees with you. What are your thoughts on whether we need to actually commit the vendor directory? The reason I ask is we were intending to not commit `vendor/` for the Elasticache broker and just run `dep ensure` and `ginkgo blah blah blah` in the integration tests. Thoughts?

[the remark this answers is not in the document]
Comment (camelpunch) [reply go/Dockerfile]: So my take is that we should remove ginkgo from here and encourage people to use the binary that corresponds to the version they've vendored. We can do that either by saying "use `go test`" or by saying "`go install` your ginkgo binary from your vendor dir". Whether or not the vendor dir is committed is a separate issue I think.

[the remark this answers is not in the document]
Comment (alext) [reply go/Dockerfile]: I agree with @camelpunch. I also think we should commit the vendor directory.

Thinking further, with the vendor directory committed, there's no need to use dep at all except when changing the dependencies. This means that we wouldn't need this container at all, and can use one of the vanilla golang containers.

The other advantage of committing the vendor directory is speed - `dep ensure` isn't quick at the moment.

**Comment by camelpunch** [comment]
@alext and I are in agreement that we should instead use a vanilla golang image, such as the one this was based upon. We can commit our vendor directories, which removes the need for git or package management tools to be on the containers.

Reproducible builds are also a big win in other ways. For example, having tests pass on one `dep ensure` and then re-`dep ensure`ing everything for a bosh release leaves the possibility for the packages to have been changed or even removed from the internet in the meantime.

**Pull request closed without merging**